backend/routes_chemicals.py

```
@@ -30,93 +31,107 @@ def decorated_function(*args, **kwargs):
-            query = Chemical.query
+        # Batch update status based on expiration and stock level to avoid N+1 queries
+        chemicals_to_update = []
+        archive_logs = []
 
-            # Filter by archived status if the column exists
+        for chemical in chemicals:
             try:
-                if not show_archived:
-                    query = query.filter(Chemical.is_archived == False)
+                is_archived = chemical.is_archived
             except:
```

Ruff (0.11.9)

Do not use bare `except`

(E722)